pgmpy/factors/CPD.py

```diff
@@ -369,6 +371,51 @@ def to_factor(self):
         return Factor(self.variables, self.cardinality, self.values)
 
 
+    def reorder_parents(self,new_order,inplace = False):
+        '''
+
+        Parameters
+        ----------
+        new_order: list
+            list of new ordering of variables
+        inplace: boolean
+            If inplace == True it will modify the CPD itself
+            otherwise new value will be returned without affecting old values
+
+        Returns
+        -------
+
+        '''
+        if not self.evidence or (set(new_order) - set(self.evidence)) or (set(self.evidence) - set(new_order)):
+            raise ValueError("New order either has missing or extra arguments")
+        else:
+            # No need to do anything if the user provides the same order
+            if new_order != self.evidence[::-1]:
+
+                # Find cardinality mapping of variables
+                card_map = dict(zip(self.evidence, self.evidence_card))
+                # Find old position mapping of variables
+                old_pos_map = dict(zip(self.evidence, range(len(self.evidence))))
+                # Create new ordering of variables based on the given order and previous order
+                trans_ord = [0]+[(old_pos_map[letter]+1) for letter in new_order][::-1]
+                # Find the actual transpose
+                new_values = np.transpose(self.values, trans_ord)
+                # Update old values if it is inplace
+                if inplace:
+                    self.evidence[:] = new_order[::-1]
+                    self.evidence_card = [card_map[var] for var in new_order[::-1]]
+                    variables = [self.variables[0]] + self.evidence
+                    cardinality = [self.variable_card] + self.evidence_card
+                    super(TabularCPD, self).__init__(variables, cardinality, new_values.flatten('C'))
+                    return self.get_cpd()
```

Comment:
@ankurankan I need your input on the way I am updating the values if it is inplace. 
I am calling __init__ method of super for updating other values but I don't really think that it is correct here.